### PR TITLE
ROC-4957: Consistent usage of free mediation value across different response models

### PR DESCRIPTION
**Pull request by singh-surjeet**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-4957

### Change description ###
Consistent usage of free mediation value across different response models

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java [hunk 1]
```diff
@@ -92,9 +93,12 @@ private boolean isReferredToJudge(ClaimantResponse response) {
 
     private boolean isRejectPartAdmitNoMediation(ClaimantResponse claimantResponse, Claim claim) {
         Response response = claim.getResponse().orElseThrow(IllegalStateException::new);
+
         return ResponseType.PART_ADMISSION.equals(response.getResponseType())
             && ClaimantResponseType.REJECTION.equals(claimantResponse.getType())
-            && !((ResponseRejection) claimantResponse).getFreeMediation().orElse(false);
+            && ((ResponseRejection) claimantResponse).getFreeMediation()
+            .filter(Predicate.isEqual(YesNoOption.NO))
+            .isPresent();
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java]: How about?
```
NO == ((ResponseRejection) claimantResponse).getFreeMediation().orElse(NO)
```

Comment (singh-surjeet) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java]: May be leave it like that, looks more java 8ishh 😄

File: src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ResponseRejectionContentProvider.java [hunk 2]
```diff
@@ -19,7 +20,9 @@
         Map<String, Object> content = new HashMap<>();
 
         responseRejection.getReason().ifPresent(reason -> content.put("rejectionReason", reason));
-        content.put("freeMediation", CCDYesNoOption.valueOf(responseRejection.getFreeMediation().orElse(false)));
+        content.put("freeMediation", CCDYesNoOption.valueOf(responseRejection
+            .getFreeMediation()
+            .orElse(YesNoOption.NO).name()));
         content.put("formNumber", DEFENCE_FORM_NO);
```
Comment (dharmendrak) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/ResponseRejectionContentProvider.java]: content.put("freeMediation", responseRejection.getFreeMediation().orElse(YesNoOption.NO).name());

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java (hunk 1 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java]: I suppose better would be `        return ResponseType.PART_ADMISSION.equals(response.getResponseType())
            && ClaimantResponseType.REJECTION.equals(claimantResponse.getType())
            && ((ResponseRejection) claimantResponse).getFreeMediation().orElse(YesNoOption.NO) != YesNoOption.YES;`

**Comment by sabahirfan** [comment]
Agree with @srineelesh comment. Please make it consistent for other places as well.

**Comment by singh-surjeet** [comment]
Scope of this story is only about consistent use of free mediation model. Story is corrected to reflect same. Other changes can done as new story with sensible sized PRs.

**Comment by singh-surjeet** [comment]
@sabahirfan @dharmendrak @srineelesh - I haven't seen review comments on codebase other than increasing scope of this story and minor if/else comments style of coding. I needed to merge this story quickly because it's blocking testing and FE PR waiting on it. Can you review other part of this PR and make your decision quickly please?

**Pull request merged**